trustregistry/tests/test_main.py

```diff
@@ -19,28 +19,221 @@ def test_create_app():
     mock_inspector = MagicMock()
+    mock_inspector.get_table_names.return_value = ["table1", "table2"]
     mock_inspect.return_value = mock_inspector
+
+    # Test the lifespan context, expecting an Exception
+    with pytest.raises(Exception, match="Migration error"):
+        async with lifespan(FastAPI()):
+            pass
+
+    # Assertions
+    mock_check_migrations.assert_called_once_with(mock_engine, mock_config.return_value)
+    mock_command.upgrade.assert_called_once_with(mock_config.return_value, "head")
+    mock_logger.info.assert_called_with("Applying database migrations...")
+    mock_logger.exception.assert_called_once_with("Error during migration")
+
+
+@pytest.mark.anyio
+@patch("trustregistry.main.engine")
+@patch("trustregistry.main.check_migrations")
+@patch("trustregistry.main.Config")
+@patch("trustregistry.main.command")
+@patch("trustregistry.main.logger")
+@patch("trustregistry.main.inspect")
+async def test_lifespan_no_migrations_needed(
+    mock_inspect,
+    mock_logger,
+    mock_command,
+    mock_config,
+    mock_check_migrations,
+    mock_engine,
+):
+    mock_check_migrations.return_value = True
+    mock_config.return_value = MagicMock()
+    mock_inspector = MagicMock()
     mock_inspector.get_table_names.return_value = ["table1", "table2"]
-    mock_connection = MagicMock()
-    mock_engine.connect.return_value.__enter__.return_value = mock_connection
+    mock_inspect.return_value = mock_inspector
 
     async with lifespan(FastAPI()):
         pass
 
-    mock_engine.dispose.assert_called_once()
-    mock_engine.connect.assert_called_once()
-    mock_inspect.assert_called_once_with(mock_connection)
+    mock_check_migrations.assert_called_once_with(mock_engine, mock_config.return_value)
+    mock_command.upgrade.assert_not_called()
+    mock_logger.info.assert_called_with("Database is up to date. No migrations needed.")
+    mock_inspect.assert_called_once_with(
+        mock_engine.connect.return_value.__enter__.return_value
+    )
     mock_inspector.get_table_names.assert_called_once()
+    mock_logger.debug.assert_called_with(
+        "TrustRegistry tables created: `{}`", ["table1", "table2"]
+    )
+
+
+@pytest.mark.parametrize(
+    "has_alembic_version,has_actors_table,current_rev,head_rev,expected",
+    [
+        (
+            False,
+            True,
+            None,
+            "head_rev",
+            False,
+        ),  # alembic_version missing, actors exists
+        (
+            False,
+            False,
+            None,
+            "head_rev",
+            False,
+        ),  # both alembic_version and actors missing
+        (
+            True,
+            True,
+            "current_rev",
+            "head_rev",
+            False,
+        ),  # alembic_version exists, revisions don't match
+        (
+            True,
+            True,
+            "same_rev",
+            "same_rev",
+            True,
+        ),  # alembic_version exists, revisions match
```

Comment:
if the comment is on the same line, then formatting expands it to this multiline view. Maybe comments can be added to line before, so it's more compact